### PR TITLE
add basic support for type_overrides

### LINKED ISSUE #41: Add a way to override types and add fields/methods in the yaml file
JS type annotations are often wrong or missing entirely, and any local fixes get overriden the next time we update the elements which causes recurring issues. If we could instead add these overrides/additions via the yaml file then they would be permanent and we wouldn't need to rely on upstream pull requests getting merged.

### PR DESCRIPTION
**Pull request by jakemac53**
closes https://github.com/dart-lang/custom-element-apigen/issues/41

### CODE REVIEW

File: lib/src/js/analyze.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ function getBehaviors(results, filePath) {
     if (behavior.contentHref != filePath) continue;
 
     behaviors.push({
-      name: behavior.is,
+      name: behavior.is.replace('Polymer.', ''),
```
Comment (sigmundch) [inline lib/src/js/analyze.js]: not sure I follow this line change

**Comment by sigmundch** [comment]
just one comment, otherwise lgtm

File: lib/src/js/analyze.js (hunk 1 above)
Comment (jakemac53) [inline lib/src/js/analyze.js]: Sometimes the behavior names start with `Polymer.`, this just strips that part out early. Previously it was being stripped out later on, and I forgot to also remove those.

Comment (sigmundch) [inline lib/src/js/analyze.js]: Ah ok, does it relate to the other changes though? or is it just a separate fix?

Comment (jakemac53) [inline lib/src/js/analyze.js]: Oh, this is what is building the json object which I am modifying later on based on the class names you provide.

Comment (jakemac53) [inline lib/src/js/analyze.js]: https://github.com/dart-lang/custom-element-apigen/pull/42/files#diff-5bfa8b8166ecc77e9cedc1dfecda5613R186

Comment (sigmundch) [inline lib/src/js/analyze.js]: Ah, I see, without this you didn't have a normalized way to refer to the polymer names.  Makes sense to me now, thank you.

**Comment by sigmundch** [comment]
still lgtm

**Pull request merged**